tests/keithley_2700.py

```diff
@@ -0,0 +1,168 @@
+import unittest
+from utils.channel_access import ChannelAccess
+from utils.ioc_launcher import get_default_ioc_dir
+from utils.test_modes import TestModes
+from utils.testing import get_running_lewis_and_ioc, skip_if_recsim
+import time
+
+DEVICE_PREFIX = "KHLY2700_01"
+
+IOCS = [
+    {
+        "name": DEVICE_PREFIX,
+        "directory": get_default_ioc_dir("KHLY2700"),
+        "macros": {},
+        "emulator": "keithley_2700",
+    },
+]
+
+TEST_MODES = [TestModes.RECSIM, TestModes.DEVSIM]
+
+
+class Status(object):
+    ON = "ON"
+    OFF = "OFF"
+
+
+class Keithley_2700Tests(unittest.TestCase):
+    """
+    Tests for the Keithley2700.
+    """
+
+    def setUp(self):
+        self._lewis, self._ioc = get_running_lewis_and_ioc("keithley_2700", DEVICE_PREFIX)
+        self.ca = ChannelAccess(default_timeout=30, device_prefix=DEVICE_PREFIX)
+        self.ca.wait_for("IDN")
+
+    def test_GIVEN_scan_state_set_WHEN_read_THEN_scan_state_is_as_expected(self):
+        sample_data = [0, 1]
+        expected_state = ["INT", "NONE"]
+        for set_value, expected_value in zip(sample_data, expected_state):
+            self.ca.assert_setting_setpoint_sets_readback(set_value, "SCAN:STATE", "SCAN:STATE:SP", expected_value)
+
+    def test_GIVEN_idn_defined_WHEN_read_THEN_idn_is_as_expected(self):
+        self.ca.assert_that_pv_is("IDN", "Keithley 2700 Emulator IDN")
+
+    def test_GIVEN_delay_state_set_WHEN_read_THEN_delay_state_is_as_expected(self):
+        sample_data = [0, 1]
+        expected_state = ["OFF", "ON"]
+        for set_value, expected_value in zip(sample_data, expected_state):
+            self.ca.assert_setting_setpoint_sets_readback(set_value, "DELAYMODE", "DELAYMODE:SP", expected_value)
+
+    def test_GIVEN_source_set_WHEN_read_THEN_source_is_as_expected(self):
+        sample_data = [0, 1, 2, 3, 4]
+        expected_channel = ["IMM", "TIM", "MAN", "BUS", "EXT"]
+        for set_value, expected_value in zip(sample_data, expected_channel):
+            self.ca.assert_setting_setpoint_sets_readback(set_value, "CONTROLSOURCE", "CONTROLSOURCE:SP",
+                                                          expected_value)
+
+    def test_GIVEN_buffer_size_set_WHEN_read_THEN_buffer_size_is_as_expected_AND_within_range(self):
+        expected_buffer_size = [5500, 0, 2, 70000]
+        alarm_state = [ChannelAccess.ALARM_NONE, ChannelAccess.ALARM_MAJOR, ChannelAccess.ALARM_NONE,
+                       ChannelAccess.ALARM_MAJOR]
```

Comment:
You appear to be testing the positive and negative cases alongside here, but that isn't indicated in the name of the test case. Please keep tests atomic.

Reply:
Seperated tests